python-oop-mini-project/account.py

```diff
@@ -0,0 +1,111 @@
+class TransactionFailedException(Exception):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+class Account():
+
+    open_metadata = {'open': ('No Display',[('opening_balance', 'Please enter a starting balance: ', 'non_negative_float')])}
+    deposit_metadata = {'deposit' : ('Deposit to account', [('deposit_amount', 'Please enter amount to deposit: ', 'non_negative_float')])}
```

Comment:
The metadata variables are used so that Bank can interact with the terminal user with arbitrary menus and prompts, and then call Account methods with the appropriate parameters.  

The open_metadata variable in Account drives the open() method in Account.   These two are inherited directly by SavingsAccount but overridden by CheckingAccount.

The deposit_metadata and deposit() method are used as inherited by both SavingsAccount and CheckingAccount.